components/Dashboard/Cards/FriendCard.js

```diff
@@ -1,39 +1,48 @@
-import React from "react";
-import { makeStyles } from "@material-ui/core";
-import { useTheme } from "@material-ui/core/styles";
-import Box from "@material-ui/core/Box";
-import Avatar from "@material-ui/core/Avatar";
-import List from "@material-ui/core/List";
-import ListItem from "@material-ui/core/ListItem";
-import ListItemText from "@material-ui/core/ListItemText";
-import ListItemAvatar from "@material-ui/core/ListItemAvatar";
+import PropTypes from 'prop-types';
+import React from 'react';
+import { makeStyles } from '@material-ui/core';
+import { useTheme } from '@material-ui/core/styles';
+import Box from '@material-ui/core/Box';
+import Avatar from '@material-ui/core/Avatar';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemText from '@material-ui/core/ListItemText';
+import ListItemAvatar from '@material-ui/core/ListItemAvatar';
+import uniqueId from 'lodash/uniqueId';
 
-const FriendCard = (props) => {
+const FriendCard = ({ isSidebarCollapsed, friendData }) => {
   const theme = useTheme();
   return (
     <Box bgcolor={theme.palette.primary.main} className="rounded-r-xl">
       <List>
-        <ListItem>
-          <ListItemAvatar>
-            <Avatar />
-          </ListItemAvatar>
-          <ListItemText primary="Yi Nan Zhang" secondary="in study session" />
-        </ListItem>
-        <ListItem>
-          <ListItemAvatar>
-            <Avatar />
-          </ListItemAvatar>
-          <ListItemText primary="Jimmy Yang" secondary="in study session" />
-        </ListItem>
-        <ListItem>
-          <ListItemAvatar>
-            <Avatar />
-          </ListItemAvatar>
-          <ListItemText primary="Charles Liu" secondary="in study session" />
-        </ListItem>
+        {isSidebarCollapsed
+          ? friendData.map(({ name, status, image }) => {
+              return (
+                <ListItem key={uniqueId(name)}>
+                  <ListItemAvatar>
+                    <Avatar />
+                  </ListItemAvatar>
+                </ListItem>
+              );
+            })
+          : friendData.map(({ name, status, image }) => {
+              return (
+                <ListItem key={uniqueId(name)}>
+                  <ListItemAvatar>
+                    <Avatar />
+                  </ListItemAvatar>
+
+                  <ListItemText primary={name} secondary={status} />
+                </ListItem>
+              );
+            })}
```

Comment:
is the only thing difffereniating the state of `isSidebarCollapsed` the line `<ListItemText primary={name} secondary={status} />`? In that case, I think you can jsut put the conditionnal around `<ListItemText primary={name} secondary={status} />`  instead of have two map functions in the ternary expression

Reply:
Tried that, it doesn't seem change state when toggling the sidebar because of how the ListItems were mapped